plugins/lookup/collection_version.py

```diff
@@ -0,0 +1,129 @@
+    description:
+      - The value to return when the collection could not be found.
+      - By default, C(none) is returned.
+    type: string
+  result_no_version:
+    description:
+      - The value to return when the collection has no version number.
+      - This can happen for collections installed from git which do not have a version number
+        in C(galaxy.yml).
+      - By default, C(*) is returned.
+    type: string
+    default: '*'
+"""
+
+EXAMPLES = """
+- name: Check version of community.general
+  ansible.builtin.debug:
+    msg: "community.general version {{ lookup('community.general.collection_version', 'community.general') }}"
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - The version number of the collections listed as input.
+      - If a collection can not be found, it will return the value provided in I(result_not_found).
+        By default, this is C(none).
+      - If a collection can be found, but the version not identified, it will value provided in
+        I(result_no_version). By default, this is C(*). This can happen for collections installed
+        from git which do not have a version number in C(galaxy.yml).
+    type: list
+    elements: str
+"""
+
+import json
+import os
+import re
+
+import yaml
+
+from ansible.errors import AnsibleLookupError
+from ansible.module_utils.compat.importlib import import_module
+from ansible.plugins.lookup import LookupBase
+
+
+FQCN_RE = re.compile(r'^[A-Za-z0-9_]+\.[A-Za-z0-9_]+$')
+
+
+def load_collection_meta_manifest(manifest_path):
+    with open(manifest_path, 'rb') as f:
+        meta = json.load(f)
+    return {
+        'version': meta['collection_info']['version'],
+    }
+
+
+def load_collection_meta_galaxy(galaxy_path, no_version='*'):
+    with open(galaxy_path, 'rb') as f:
+        meta = yaml.safe_load(f)
+    return {
+        'version': meta.get('version') or no_version,
+    }
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        result = []
+        self.set_options(var_options=variables, direct=kwargs)
+        not_found = self.get_option('result_not_found')
+        no_version = self.get_option('result_no_version')
+
+        for term in terms:
+            if not FQCN_RE.match(term):
+                raise AnsibleLookupError('"{term}" is not a FQCN'.format(term=term))
+
+            try:
+                collection_pkg = import_module('ansible_collections.{fqcn}'.format(fqcn=term))
+                if not collection_pkg:
+                    raise Exception('collection not found')
+            except Exception:
+                # Collection not found
+                result.append(not_found)
+                continue
+
+            try:
+                path = os.path.dirname(collection_pkg.__file__)
+                manifest_path = os.path.join(path, 'MANIFEST.json')
+                if os.path.exists(manifest_path):
+                    data = load_collection_meta_manifest(manifest_path)
```

Comment:
Would it be a bit nicer to have a `continue` as the following line here since we can then drop the `else` and reduce the indent by one level for the rest of the code?

Reply:
Hmm, that won't work without also replicating the `result.append(data.get('version', no_version))` line. I think I'd rather move this code into a method, then it can simply return `data` when it found some.

Reply:
Refactored in 27a5864256aecfc6c57ace85048202cb2b072f5f.